helix-core/src/main/java/org/apache/helix/controller/changedetector/ResourceChangeSnapshot.java

```diff
@@ -72,6 +73,17 @@
     _changedTypes = new HashSet<>(dataProvider.getRefreshedChangeTypes());
     _instanceConfigMap = new HashMap<>(dataProvider.getInstanceConfigMap());
     _idealStateMap = new HashMap<>(dataProvider.getIdealStates());
+    for (String resourceName : _idealStateMap.keySet()) {
+      IdealState orgIdealState = _idealStateMap.get(resourceName);
+      if (orgIdealState.getRebalanceMode().equals(IdealState.RebalanceMode.FULL_AUTO)) {
+        IdealState trimmedIdealState = new IdealState(orgIdealState.getRecord());
+        // For FullAuto resources, map fields and list fields in the IdealStates is not user's input.
+        // So there is no need to detect any change in these 2 scopes.
```

Comment:
Nit: these 2 scopes -> this would technically be "these two fields"?

Reply:
Good point.